### ISSUE TITLE
CMake: C/C++ debug optimization can be further improved

### ISSUE THREAD

**Issue by stephengtuggy** [vegastrike/Vega-Strike-Engine-Source#173]
Looking at the `gcc` online docs for [debugging options](https://gcc.gnu.org/onlinedocs/gcc/Debugging-Options.html) and [optimization options](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html#Optimize-Options), it looks like the best combination is `-Og -g3`.

Indeed, when debugging the result using `gdb`, there does seem to be more debugging information available.

Let's add `-Og -g3` to the DEBUG configuration in CMakeLists.txt, for the C and C++ compilers. If there are no objections, that is.

**Comment by LifWirser** [comment]
For GDB (gnome dubugger) and other such apps, break points ( effective stop sign for debugging). After giving the matter some consideration my suggestion would be after each "section is initialized (python,GL,audio,etc...).
 Also the need for accurate code listing greatly helps. For example doxygen  as well as other code "profiler" needs a script to tell it what to document - note both VS-engine and Assests need to be availabel.
I have so far managed to get "hello.py to run via gdb-py (package not file name), The issue with breakpoints is relevent for gdb to properly access VS-py code. I considered a module to setup needed paths etc- butr if there is an error with VS module you would not catch it, . So a "hard coded breakpoint right after VS is read by python allows in theroy access to python as VSE engine sees it.
 Just The scripts for Gdb/doxygen  Are a project in it self. Let alone adding The resr of available os's. Also both of these examples cross two seperat paths not including future mods etc.

Bottom line , the need to be able to better control over execution is clear , the benifits for one achiving 1.0.0 quicker is just one

**Comment by BenjamenMeyer** [comment]
> For GDB (gnome dubugger) 

GDB is the GNU Debugger

> and other such apps, break points ( effective stop sign for debugging). After giving the matter some consideration my suggestion would be after each "section is initialized (python,GL,audio,etc...). Also the need for accurate code listing greatly helps.

> For example doxygen as well as other code "profiler" needs a script to tell it what to document - note both VS-engine and Assests need to be availabel.

No. The Assets and Engine are separate things, and we need to be able to document them separately. We also need to develop some basic test framing assets for use in Unit Tests and specific test scenarios - not a complete game.

> I have so far managed to get "hello.py to run via gdb-py (package not file name), The issue with breakpoints is relevent for gdb to properly access VS-py code. I considered a module to setup needed paths etc- butr if there is an error with VS module you would not catch it, . So a "hard coded breakpoint right after VS is read by python allows in theroy access to python as VSE engine sees it. Just The scripts for Gdb/doxygen Are a project in it self. Let alone adding The resr of available os's. Also both of these examples cross two seperat paths not including future mods etc.

sounds like something we may be able to incorporate into a basic test assets infrastructure.
 
> Bottom line , the need to be able to better control over execution is clear , the benifits for one achiving 1.0.0 quicker is just one

We've got a long way to the `1.0.0` mark; but I'm not hung up on version numbers so much. We've made a lot of progress in the last couple years with getting things updated and slowly getting things in a better state, especially considering the size of the team doing the work. We'll get there.
